drivers/spi/spi_rpi_pico_pio.c

```diff
@@ -41,44 +44,157 @@ struct spi_pico_pio_data {
 	uint32_t rx_count;
 	PIO pio;
 	size_t pio_sm;
+	uint32_t pio_tx_offset;
+	uint32_t pio_rx_offset;
+	uint32_t pio_rx_wrap_target;
+	uint32_t pio_rx_wrap;
+	uint32_t tx_period_ticks;
+	uint32_t bits;
 	uint32_t dfs;
 };
 
-RPI_PICO_PIO_DEFINE_PROGRAM(spi_cpol_0_cpha_0, 0, 1,
+/* ------------ */
+/* spi_mode_0_0 */
+/* ------------ */
+
+#define SPI_MODE_0_0_WRAP_TARGET 0
+#define SPI_MODE_0_0_WRAP        1
```

Comment:
These lines are bit verbose.
These only use one time, I think no need to define.

Reply:
I was trying to create a consistent pattern across all the PIO blocks that works with the current Zephyr PIO definitions.  I'd prefer to leave these.

Reply:
I got it. Let's do as so.